### PR TITLE
[oauth2-proxy] Update oauth2-proxy to 7.0.0

### PR DESCRIPTION
**Pull request by jmmaloney4**
Update the image used in oauth2-proxy to `v7.0.1` from `v6.1.1`.

Updated image.

Various improvements.

A few breaking changes in the configuration file. See [https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.0.0](https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.0.0).

- fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [ ] (optional) Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [x] (optional) Variables are documented in the README.md

### CODE REVIEW

**Comment by billimek** [comment]
@jmmaloney4 there appears to be a minor [linting issue](https://github.com/k8s-at-home/charts/pull/619/checks?check_run_id=1964460149) with this change - can you please take a look and push a change?

**Comment by bjw-s** [comment]
Also I think the README update should go in to one of the other files. On mobile right now, can't quickly check which one.

**Comment by onedr0p** [comment]
We need to merge https://github.com/k8s-at-home/charts/pull/620 and this needs to be rebased.

File: charts/oauth2-proxy/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: oauth2-proxy
-version: 4.4.0
+version: 5.0.0
 apiVersion: v1
-appVersion: 5.1.0
+appVersion: "v7.0.1"
```
Comment (onedr0p) [inline charts/oauth2-proxy/Chart.yaml:4]:
```suggestion
appVersion: 7.0.1
```

File: charts/oauth2-proxy/values.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ config:
 
 image:
   repository: "quay.io/oauth2-proxy/oauth2-proxy"
-  tag: "v6.1.1"
+  # tag: "v6.1.1"
```
Comment (onedr0p) [inline charts/oauth2-proxy/values.yaml:33]:
```suggestion
  tag: "v7.0.1"
```

**Review by onedr0p — CHANGES_REQUESTED** [review]

**Comment by onedr0p** [comment]
Thanks for updating!

**Comment by jmmaloney4** [comment]
Thanks for your patience with my first pull request!

**Review by onedr0p — APPROVED** [review]

**Pull request merged**